addons/OPT/GPS/fn_addUnitToGPS.sqf

```diff
@@ -22,99 +20,100 @@
 * No
 * 
 * Example:
-* _iconID = [player, "Icon_Player"] call FUNC(addUnitToGPS);
+* [player, "Icon_Player"] call FUNC(addUnitToGPS);
 */
 
 #include "macros.hpp"
 
 
-params ["_newUnit", "_iconId"];
 
-// Decide on the right color
-private _color = if (CLib_Player isEqualTo _newUnit) then {
-    COLOR_PLAYER_UNIT
-    } else {
-        if (group CLib_Player isEqualTo group _newUnit) then {
-            COLOR_OWN_GROUP
-        } else {
-            COLOR_SIDE
-        }
+[{
+    params ["_newUnit"];
+    DUMP("UNIT ICON ADDED");
+    DUMP(_newUnit);
+
+    if (!([_newUnit] call FUNC(isUnitVisible))) exitWith {
+        DUMP(format ["Unit %1 is not visible", _newUnit]);
```

Comment:
Ganz schlechte Idee. Das macht wahrscheinlich nicht das, was du vllt willst. Man darf keine Array-Literals in Makro-Argumenten verwenden (weil da Kommas drin sind). Du kannst das entweder so lösen:
```
#define LOG_ARG format ["Unit %1 is not visible", _newUnit]
DUMP(LOG_ARG);
```
ode du verzichtest auf das `format` und baust den String ganz traditionell via `+` zusammen:
```
DUMP("Unit " + str _newUnit + " is not visible");
```

Reply:
Hm, hatte bisher keine Probleme damit. Meine Vermutung: da das Makro erst auf dem Ergebnis des format-Befehls arbeitet, gibt's keinen Ärger, weil das Makro dann nur mehr einen langweiligen String verarbeitet.

Reply:
So funktionieren Makros nicht - die werden ja vom Preprocessor verarbeitet. Der wartet also nicht, darauf, dass irgendein SQF-Befehl fertig ist. Ich hatte das so auch mal, aber es hat halt nicht funktioniert (aus genanntem Grund). Ganz sicher dass du diese Log-Nachricht schonmal wie erwartet im Log gefunden hast?